src/Plugin/Block/SharethisBlock.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\sharethis\Plugin\Block\AutologoutWarningBlock.
+ */
+
+namespace Drupal\sharethis\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Config\Config;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides an 'Sharethis Logout info' block.
+ *
+ * @Block(
+ *   id = "sharethis_block",
+ *   admin_label = @Translation("Sharethis"),
+ * )
+ */
+class SharethisBlock extends BlockBase implements ContainerFactoryPluginInterface {
+  /**
+   * Constructs an SharethisBlock object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   *   The module manager service.
+   * @param \Drupal\Core\Config\Config $sharethis_settings
+   *   The config object for 'sharethis.settings'.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, Config $sharethis_settings) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->sharethisSettings = $sharethis_settings;
```

Comment:
please add a protected datamember to class to hold the sharethis settings 

protected $sharethisSettings